### PR TITLE
Update swingset-runner tests for new Zoe

### PR DESCRIPTION
**Pull request by michaelfig**
This takes a fresh copy of `swingset-runner`s Zoe tests from `zoe-release-candidate`, and applies the changes described in https://github.com/Agoric/agoric-sdk/issues/1194

Please see
https://github.com/Agoric/agoric-sdk/pull/1487/files/c5885700..93be4a9b
to review those changes in isolation.

Many of the Zoe typing changes were moved to #1488.

#dapp-encouragement-branch: new-zoe

### CODE REVIEW

File: packages/zoe/src/contracts/mintAndSellNFT.js [hunk 1]
```diff
@@ -85,14 +85,15 @@ const start = zcf => {
           .then(sellItemsCreatorSeat => {
             return harden({
               sellItemsCreatorSeat,
-              sellItemsCreatorFacet: creatorFacet,
+              sellItemsCreatorFacet: /** @type {SellItemsCreatorFacet} */ (creatorFacet),
```
Comment (dtribble) [inline packages/zoe/src/contracts/mintAndSellNFT.js]: I'm not fond of having to mix type goo into our code. can this be moved out of the method code?

**Review by dtribble — APPROVED** [review]

File: packages/swingset-runner/demo/zoeTests/swingset.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "bootstrap": "bootstrap",
   "bundles": {
     "zcf": {
-      "sourceSpec": "../../../zoe/src/contractFacet.js"
+      "sourceSpec": "@agoric/zoe/src/contractFacet/contractFacet.js"
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/swingset-runner/demo/zoeTests/swingset.json]: In #1488.

File: packages/swingset-runner/demo/zoeTests/vat-alice.js [hunk 3]
```diff
@@ -1,12 +1,23 @@
...
+import '@agoric/zoe/src/contracts/exported';
+
 const log = makePrintLog();
 
+/**
+ * @param {ZoeService} zoe
+ * @param {Issuer[]} issuers
+ * @param {Payment[]} payments
+ * @param {Record<string,Installation>} installations
+ * @param {*} timer
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/swingset-runner/demo/zoeTests/vat-alice.js]: It means`any` but I use it specifically when I haven't bothered typing something.

File: packages/zoe/src/contracts/mintAndSellNFT.js (hunk 1 above)
Comment (michaelfig) [reply packages/zoe/src/contracts/mintAndSellNFT.js]: It is in #1488

**Comment by michaelfig** [comment]
#1488 has many of the typing changes.  The current PR will be force-pushed just to affect swingset-runner/demo.

File: packages/zoe/test/swingsetTests/zoe-metering/infiniteTestLoop.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-export const start = zcf => {
+export const start = _zcf => {
```
Comment (FUDCo) [inline packages/zoe/test/swingsetTests/zoe-metering/infiniteTestLoop.js:1]: Good change but note that this one is not in `swingset-runner`.

File: packages/swingset-runner/demo/encouragementBot/bootstrap.js [hunk 5]
```diff
@@ -1,5 +1,3 @@
-/* global harden */
```
Comment (FUDCo) [inline packages/swingset-runner/demo/encouragementBot/bootstrap.js:1]: I'm thrilled to have this go away!

File: packages/swingset-runner/demo/zoeTests/vat-zoe.js [hunk 6]
```diff
@@ -1,9 +1,8 @@
-/* global harden */
-
+// noinspection ES6PreferShortImport
 import { makeZoe } from '@agoric/zoe';
 
-export function buildRootObject(_vatPowers, vatParameters) {
+export function buildRootObject(__vatPowers) {
```
Comment (FUDCo) [inline packages/swingset-runner/demo/zoeTests/vat-zoe.js]: Double underscore here looks to me like a typo.  If `vatParameters` is no longer being used, probably should just take the arg out entirely.  On the other hand, why is `vatParameters` no longer being used?  How is the bundle name getting to Zoe?  If the bundle name is baked into Zoe, that's arguably dangerous.  If it's not using named bundles, then something got lost in the Zoe updates and needs to be fixed.

**Review by FUDCo — CHANGES_REQUESTED** [review]
Looks mostly good with one exception (well, aside from one very minor nit in `zoeTests/bootstrap.js`):  I suspect the dropping of the bundle name handling in `zoeTests/vat-zoe.js` is a mistake (especially since the same did not happen to `exchangeBenchmark`).  This might be a false alarm on my part if you've refactored `makeZoe` (in which case `exchangeBenchmark` needs revisiting) but if it's what I think it is it's a serious booboo that should be corrected.

File: packages/swingset-runner/demo/zoeTests/vat-zoe.js (hunk 6 above)
Comment (michaelfig) [reply packages/swingset-runner/demo/zoeTests/vat-zoe.js]: I investigated, and the bundle name is neither baked in, nor using named bundles.  Zoe in the `zoe-release-candidate` branch uses an import of the pre-bundled contractFacet which is built by the zoe package's `yarn build` step.  Maybe we want that to be something different, (in which case I'd suggest creating an issue) but `makeZoe` takes only one argument.

In either case, it's out of scope for this PR.

Comment (FUDCo) [reply packages/swingset-runner/demo/zoeTests/vat-zoe.js]: No, those changes to `makeZoe` were merged into the master branch earlier.  If your release candidate doesn't have them, something got messed up in the merge of the RC with master and absolutely needs to be fixed.

**Review by FUDCo — CHANGES_REQUESTED** [review]
Something got lost in the RC merge.  `makeZoe` should take two parameters, the second of which is the optional zcf bundle name.

File: packages/swingset-runner/demo/zoeTests/vat-zoe.js (hunk 6 above)
Comment (michaelfig) [reply packages/swingset-runner/demo/zoeTests/vat-zoe.js]: > No, those changes to `makeZoe` were merged into the master branch earlier. If your release candidate doesn't have them, something got messed up in the merge of the RC with master and absolutely needs to be fixed.

Fixed in #1504

**Pull request merged**